CRM/Contact/Page/AJAX.php

```diff
@@ -1017,13 +1017,9 @@ public static function getAddressDisplay() {
    * Mark dupe pairs as selected from un-selected state or vice-versa, in dupe cache table.
    */
   public static function toggleDedupeSelect() {
-    $rgid = CRM_Utils_Type::escape($_REQUEST['rgid'], 'Integer');
-    $gid  = CRM_Utils_Type::escape($_REQUEST['gid'], 'Integer');
     $pnid = $_REQUEST['pnid'];
     $isSelected = CRM_Utils_Type::escape($_REQUEST['is_selected'], 'Boolean');
-    $criteria = CRM_Utils_Request::retrieve('criteria', 'Json', $null, FALSE, '{}');
```

Comment:
So this function wasn't working when criteria was passed because the smarty layer escaping was leaving it empty & the cachekey couldn't be calculated. Turns out the criteria is really only needed to calculate the cacheKey so instead just pass that & simplify. Note the very last line on this PR's changes is where we change what is passed into this PR